packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -312,14 +315,70 @@ export class PHPRequestHandler<PHP extends BasePHP> {
 			),
 			this.rewriteRules
 		);
-		const fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
-		if (!seemsLikeAPHPRequestHandlerPath(fsPath)) {
-			return this.#serveStaticFile(
-				await this.processManager.getPrimaryPhp(),
-				fsPath
-			);
+
+		const primaryPhp = await this.getPrimaryPhp();
+
+		let fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
+
+		// We can only satisfy requests for directories with a default file
+		// so let's first resolve to a default path when available.
+		if (primaryPhp.isDir(fsPath)) {
```

Comment:
Let's add a more extensive note explaining how this is default WP behavior - it's not obvious this logic is the right one.

Reply:
> Let's add a more extensive note explaining how this is default WP behavior - it's not obvious this logic is the right one.

Good point! Added this:
```ts
// Ensure directory URIs have a trailing slash. Otherwise,
// relative URIs in index.php or index.html files are relative
// to the next directory up.
//
// Example:
// For a request to "/wp-admin", the relative link "edit.php"
// resolves to "/edit.php" rather than "/wp-admin/edit.php".
//
// This is correct behavior for the browser:
// https://www.rfc-editor.org/rfc/rfc3986#section-5.2.3
//
// But the intent for `/wp-admin/index.php` is that its relative
// URIs are relative to `/wp-admin/`.
//
// In fact, WordPress also redirects like this when given a chance.
// - https://github.com/WordPress/wordpress-develop/blob/b6a3b9c7d1ce33cbeca6f95871a26c48141e524b/src/wp-includes/canonical.php#L696
// - https://github.com/WordPress/wordpress-develop/blob/b6a3b9c7d1ce33cbeca6f95871a26c48141e524b/src/wp-includes/canonical.php#L1036-L1045
// - https://github.com/WordPress/wordpress-develop/blob/b6a3b9c7d1ce33cbeca6f95871a26c48141e524b/src/wp-includes/link-template.php#L3558
```